---
title: "Review_Work"
output: html_document
date: "2025-04-04"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,message=F, warning=TRUE)
```


```{r}
#LOAD LIBRARIES
pacman::p_load(dplyr,ggplot2,grid,ggpubr) 

path='../data/'
data=read.table(paste0(path,'supplementary_figure_1_xx.tsv'), header=T, na.strings = "NA")

#ORDER THE GENE NAMES
data$Gene <- factor(replace(as.character(data$Gene), is.na(data$Gene), "NA"),
                    levels = c('PB2','PB1','PA','HA','NP','NA','NS','MP'))

#SPECIFY COLOR PATTERNS
colors <- c("#4daf4a","#d7301f","#009E73",'#e7298a',"#0072B2","#D55E00","#CC79A7","#7a2c1d")

#FUNCTION TO PLOT FIGURE
generate_figure <- function(data_frame,limit_length,max_value,min_value){
  fig = ggplot(data_frame, aes(x=Position, y=depth_log10,color=dilution)) +
    facet_wrap(~Gene, nrow = 8,scales = "free_x")+
    scale_y_continuous(limits=c(min_value,max_value),breaks=seq(min_value,max_value,by=1)) +
    scale_x_continuous(breaks = c(1,1000,2000),limits=c(1,limit_length)) +
    labs(y = "Sequencing Depth (log10)", x = "Nucleotide position") +
    scale_color_manual(values = colors) +
    geom_line(stat='identity',size=1) + theme_classic() +
    theme(axis.text = element_text(size=12,color='black'),
          axis.title.y = element_blank(),
          axis.title = element_text(size=12,color='black'),
          strip.text = element_text(size = 12, color='black'),
          legend.key.size = unit(0.9, "cm"),legend.title = element_blank(),
          plot.margin = margin(30, 10, 10, 10),
          legend.text = element_text(size=12))
  return(fig)
}

#LIST GENE BY LENGTH
short_genes <- c("NP","NA","NS","MP")
long_genes <- c("HA","PA","PB1","PB2")

#determine min and max values for the two gene categories
short_data_min <- floor(data %>% filter(Gene %in% short_genes) %>% pull(depth_log10) %>% min())
short_data_max <- ceiling(data %>% filter(Gene %in% short_genes) %>% pull(depth_log10) %>% max())
long_data_min <- floor(data %>% filter(Gene %in% long_genes) %>% pull(depth_log10) %>% min())
long_data_max <- ceiling(data %>% filter(Gene %in% long_genes) %>% pull(depth_log10) %>% max())


#FUNCTION GENERATE FIGURES FOR THE SHORT AND LONG GENE SEGEMENTS
coverage_plot <- function(condition_name){
  df = data %>% filter(condition == condition_name) %>% filter(depth_log10 >0)
  short_genes <- df %>% filter(Gene %in% short_genes)
  long_genes <- df %>% filter(Gene %in% long_genes)
  fig_long_df <- generate_figure(long_genes,2300,long_data_max,long_data_min)
  fig_short_df <- generate_figure(short_genes,1000,short_data_max,short_data_min)
  return(list(long=fig_long_df, short=fig_short_df))
}

#RUN THE COVERAGE FUNCTION
L5_plot <- coverage_plot('Licheri_5uM')
Z2_plot <- coverage_plot('Zhou_2P')
Z3_plot <- coverage_plot('Zhou_3P')
R3_plot <- coverage_plot('Rambo_3P')


#MERGE FIGURES FOR THE LONGER GENES 
fig_long = ggarrange(Z2_plot$long,Z3_plot$long,R3_plot$long,L5_plot$long,nrow=1,
                common.legend = T,legend='bottom',label.x=0,
                labels=c('Zhou et al. 09','Zhou et al. 12','Rambo et al','Current Study'),
                font.label=list(size=10,face='bold'))
figure = annotate_figure(fig_long, left = textGrob("Sequencing Depth (log10)", rot = 90,
                                                 vjust = 1, gp = gpar(cex=0.8)))
ggsave(paste0(path,'../Figures/supplementary_figure_1_xxa.pdf'),plot= figure,dpi=600, width=210, height=200,units="mm")


#MERGE FIGURES FOR THE SHORTER GENES 
fig_short = ggarrange(Z2_plot$short,Z3_plot$short,R3_plot$short,L5_plot$short,nrow=1,
                common.legend = T,legend='bottom',label.x=0,
                labels=c('Zhou et al. 09','Zhou et al. 12','Rambo et al','Current Study'),
                font.label=list(size=10,face='bold'))
figure = annotate_figure(fig_short, left = textGrob("Sequencing Depth (log10)", rot = 90,
                                                 vjust = 1, gp = gpar(cex =1.1)))
ggsave(paste0(path,'../Figures/supplementary_figure_1_xxb.pdf'),plot= figure,dpi=600, width=210, height=200,units="mm")

```


