---
title: "Review_Work"
output: html_document
date: "2025-04-04"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r,message=F, warning=TRUE}
library(ggplot2)
library(dplyr)
library(grid)
library(ggpubr)

```




```{r}
path = './1_QC_Assembly/'
data=read.table(paste0(path, '/minimap_coverage_data.tsv'), header=T, na.strings = "NA")
#drop Licheri
data <- data %>% filter(condition != 'Licheri_2uM')
data$gene <- factor(replace(as.character(data$gene), is.na(data$gene), "NA"),
                    levels = c('PB2','PB1','PA','HA','NP','NA','NS','MP'))

colors <- c("#4daf4a","#d7301f","#009E73",'#e7298a',"#0072B2","#D55E00","#CC79A7","#7a2c1d")

axis_text_size=12

generate_figure <- function(data_frame,limit_length,max_value,min_value){
  data_frame$gene <- factor(replace(as.character(data_frame$gene), is.na(data_frame$gene), "NA"),
                    levels = c('PB2','PB1','PA','HA','NP','NA','NS','MP'))
  fig = ggplot(data_frame, aes(x=position, y=depth_log10,color=dilution)) +
    facet_wrap(~gene, nrow = 8,scales = "free_x")+
    scale_y_continuous(limits=c(min_value,max_value),breaks=seq(min_value,max_value,by=1)) +
    scale_x_continuous(breaks = seq(1, 2500,1000))+#  breaks = c(1,1000,2000),limits=c(1,limit_length)) +
    labs(y = "Sequencing Depth (log10)", x = "Genome position") +
    scale_color_manual(values = colors) +
    geom_line(stat='identity',size=1) + theme_classic() +
    theme(axis.text = element_text(size=axis_text_size,color='black'),
          axis.title.y = element_blank(),
          strip.background = element_blank(),
          axis.title = element_text(size=axis_text_size,color='black'),
          strip.text = element_text(size = axis_text_size, color='black', face='bold'),
          legend.key.size = unit(0.9, "cm"),legend.title = element_blank(),
          plot.margin = margin(30, 10, 10, 10),
          legend.text = element_text(size=12)) +
      guides(colour = guide_legend(nrow = 1))
  return(fig)
}

short_genes <- c("NP","NA","NS","MP")
long_genes <- c("HA","PA","PB1","PB2")

short_data_min <- floor(data %>% filter(gene %in% short_genes) %>% pull(depth_log10) %>% min())
short_data_max <- ceiling(data %>% filter(gene %in% short_genes) %>% pull(depth_log10) %>% max())

long_data_min <- floor(data %>% filter(gene %in% long_genes) %>% pull(depth_log10) %>% min())
long_data_max <- ceiling(data %>% filter(gene %in% long_genes) %>% pull(depth_log10) %>% max())

coverage_plot <- function(condition_name){
  df = data %>% filter(condition == condition_name) %>% filter(depth_log10 >0)
  short_genes <- df %>% filter(gene %in% short_genes)
  long_genes <- df %>% filter(gene %in% long_genes)
  fig_long_df <- generate_figure(long_genes,2300,long_data_max,long_data_min)
  fig_short_df <- generate_figure(short_genes,1000,short_data_max,short_data_min)
  return(list(long=fig_long_df, short=fig_short_df))
}

L5_plot <- coverage_plot('Licheri_5uM')
Z2_plot <- coverage_plot('Zhou_2P')
Z3_plot <- coverage_plot('Zhou_3P')
R3_plot <- coverage_plot('Rambo_3P')


fig_long = ggarrange(Z2_plot$long,Z3_plot$long,R3_plot$long,L5_plot$long,nrow=1,
                common.legend = T,legend='bottom',label.x=0,
                labels=c('Zhou et al. 2009','Zhou et al. 2012','Rambo et al. 2020','Current protocol'),
                font.label=list(size=10,face='bold'))
figure_long = annotate_figure(fig_long, left = textGrob("Average Sequencing Depth (log10)", rot = 90,
                                                 vjust = 1, gp = gpar(cex=0.8)))
ggsave(paste0(path, 'minimap_coverage_subsampled_long_C.pdf'),plot= figure_long,dpi=600, width=210, height=200,units="mm")


fig_short = ggarrange(Z2_plot$short,Z3_plot$short,R3_plot$short,L5_plot$short,nrow=1,
                common.legend = T,legend='bottom',label.x=0,
                labels=c('Zhou et al. 2009','Zhou et al. 2012','Rambo et al. 2020','Current protocol'),
                font.label=list(size=10,face='bold'))
figure = annotate_figure(fig_short, left = textGrob("Average Sequencing Depth (log10)", rot = 90,
                                                 vjust = 1, gp = gpar(cex =1.1)))
ggsave(paste0(path,'minimap_coverage_subsampled_short_D.pdf'),plot= figure,dpi=600, width=210, height=200,units="mm")

new_figure <- ggarrange(fig_long, fig_short, nrow=2, common.legend = T, align='hv')
ggsave(paste0(path,'minimap_coverage_subsampled_short_long_merged.pdf'),plot= new_figure,dpi=600,  width=250, height=400,units="mm")


```
